### PR TITLE
INT-15: Update permission settings to work properly

### PR DESCRIPTION
**Pull request by whisk3y**
Fixed some typos in the user permission settings and added the request to get the UserPermissionReport which was previously not populating user permissions.

### CODE REVIEW

File: dyn/tm/services/dsf.py [hunk 1]
```diff
@@ -2179,7 +2179,7 @@ def _post(self, service_id, publish=True):
         """
         self.service_id = service_id
         uri = '/DSFResponsePool/{}/'.format(self.service_id)
-        api_args = {'publish': 'Y', 'label': self._label,
+        api_args = {'publish': 'N', 'label': self._label,
```
Comment (Marchowes) [inline dyn/tm/services/dsf.py]: I doesn't really matter functionally (in this case),  but you can just exclude the 'publish' :'N' from this line. I'd hazard a guess there are probably more of these in this file.

File: dyn/tm/accounts.py [hunk 2]
```diff
@@ -750,6 +764,42 @@ def delete_permissions_group(self, group):
...
+        """Add individual zones to this :class:`~dyn.tm.accounts.User`
+
+        :param zone: the zone to add
+        """
+        if zone not in self._zone:
+            self._zone.append(zone)
+            uri = '/UserZoneEntry/{}/{}/'.format(self._user_name, zone)
+            DynectSession.get_session().execute(uri, 'POST')
+
+    def replace_zone(self, zones):
```
Comment (Marchowes) [inline dyn/tm/accounts.py]: Doesn't look like this actually sends anything to the API. Also, docstring might not be right? (or at least, its unclear)...

File: dyn/tm/accounts.py [hunk 3]
```diff
@@ -435,6 +437,17 @@ def _get(self):
...
+        response = DynectSession.get_session().execute(self._permission_report_uri, 'POST',
+                                                       api_args)
+
+        for val in response['data']['allowed']:
+            print(val)
+            self.permissions.append(val['name'])
+            for zone in val['zone']:
+                if zone['zone_name'] not in self._zone:
+                    self._zone.append(zone['zone_name'])
+
```
Comment (Marchowes) [inline dyn/tm/accounts.py]: This whole block is just here to update the list of permissions after every get right? you might want to make it into its own function and call it on _get. Also, remove the print.

File: dyn/tm/accounts.py [hunk 4]
```diff
@@ -618,7 +631,7 @@ def permission(self):
         """A list of permissions assigned to this
         :class:`~dyn.tm.accounts.User`
         """
-        return self._permission
+        return self.permissions
```
Comment (Marchowes) [inline dyn/tm/accounts.py]: keep this as _permission and change ln 446

**Comment by whisk3y** [comment]
@Marchowes @moogar0880 Can I get a PR Review?

File: dyn/tm/accounts.py [hunk 5]
```diff
@@ -443,12 +444,33 @@ def _get(self):
         for key, val in response['data'].items():
             setattr(self, '_' + key, val)
 
+        self._get_permissions(self)
```
Comment (Marchowes) [inline dyn/tm/accounts.py]: self._get_permissions()

File: dyn/tm/accounts.py [hunk 6]
```diff
@@ -408,6 +409,18 @@ def _post(self, password, email, first_name, last_name, nickname,
         """Create a new :class:`~dyn.tm.accounts.User` object on the DynECT
         System
         """
+
+        api_args = {'password': password, 'email': email,
```
Comment (Marchowes) [inline dyn/tm/accounts.py]: `permissions` needs to be `permission` and (sadly/tediously) all these values? Yep, they're going to have to be added conditionally. That is, if they are set, send them, if not, don't.

**Comment by whisk3y** [comment]
@Marchowes was this good to go after your last check?

**Comment by Marchowes** [comment]
This passes our Permission checks, which I know an earlier version did not. I'm fairly certain this is good. I know when I was updating it I did a fair bit of testing

**Pull request merged**